base/bundle.cpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "base/bundle.hpp"
+
+#include <functional>
+#include <queue>
+
+#include "base/map_util.hpp"
+#include "base/status.hpp"
+
+namespace principia {
+namespace base {
+
+Bundle::Bundle(int threads) {
+  {
+    std::unique_lock<std::mutex> l(lock_);
+    joining_ = false;
+  }
+  for (int i = 0; i < threads; ++i) {
+    threads_.emplace_back([this]() {
+      std::shared_ptr<Task> current_task;
+      for (;;) {
+        {
+          std::unique_lock<std::mutex> l(lock_);
+          if (joining_ && tasks_.empty()) {
+            return;
+          }
+          tasks_not_empty_or_joining_.wait(
+              l,
```

Comment:
It took me a while to figure out that this is not `1`.  Maybe name the `unique_lock` `ul`.  Stupid language.